matchzoo/preprocessors.py

```diff
@@ -0,0 +1,133 @@
+"""Matchzoo toolkit for text pre-processing."""
+
+import re
+import nltk
+import jieba
+import many_stop_words
+from functools import reduce
+
+
+def get_stopwords(lang: str='en') -> list:
+    """
+    Get stopwords based on language.
+
+    :params lang: language code.
+
+    :return stop_list: list of stop words.
+    """
+    return many_stop_words.get_stop_words(lang)
+
+
+def tokenizer(raw: str) -> list:
+    """
+    Word tokenization.
+
+    :param raw: raw input given by user.
+
+    :return terms: list of tokenized terms.
+    """
+    return nltk.word_tokenize(raw)
+
+
+def segmentation(raw: str) -> list:
+    """
+    Chinese sentence segmentation.
+
+    :param raw: raw input given by user.
+
+    :return terms: list of segmentated terms.
+    """
+    return list(jieba.cut(raw))
+
+
+def to_lowercase(terms: list) -> list:
+    """
+    Convert list of terms to lower case.
+
+    :param terms: list of terms.
+
+    :return terms: lower-cased list of terms.
+    """
+    return [term.lower() for term in terms]
+
+
+def remove_stopwords(terms: list, lang: str='en') -> list:
+    """
+    Remove stopwords from list of tokenized terms.
+
+    :param terms: list of tokenized terms.
+    :param lang: language code for stopwords.
+
+    :return terms: list of tokenized terms without stopwords.
+    """
+    return [term for term in terms if term not in get_stopwords(lang)]
+
+
+def remove_punctuation(terms: list) -> list:
+    """
+    Remove punctuations from list of terms.
+
+    :param terms: list of
+
+    :return rv: terms  without punctuation.
+    """
+    rv = []
+    for term in terms:
+        term = re.sub(r'[^\w\s]', '', term)
+        if term != '':
+            rv.append(term)
+    return rv
```

Comment:
Maybe the whole thing could be written as something like `list(filter(lambda x: re.sub(r'[^\w\s]', '', x), terms))` but anyways this is clear enough.